src/SearchField/SearchFieldAdvanced.jsx

```diff
@@ -0,0 +1,176 @@
+            handleBlur,
+            handleChange,
+            refs: {
+              input: inputRef,
+              submitButton: submitButtonRef,
+            },
+          }}
+        >
+          {children}
+        </SearchFieldContext.Provider>
+      </form>
+    </div>
+  );
+};
+
+SearchFieldAdvanced.propTypes = {
+  /** specifies the nested child elements. At a minimum, `SearchField.Label` and `SearchField.Input` are required. */
+  children: PropTypes.node.isRequired,
+  /** specifies a callback function for when the `SearchField` is submitted by the user. For example:
+  ```jsx
+  <SearchField onSubmit={(value) => { console.log(value); } />
+  ``` */
+  onSubmit: PropTypes.func.isRequired,
+  /** specifies a custom class name. */
+  className: PropTypes.string,
+  /** specifies a callback function for when the user loses focus in the `SearchField` component. The default is an empty function. For example:
+  ```jsx
+  <SearchField onBlur={event => console.log(event)} />
+  ``` */
+  onBlur: PropTypes.func,
+  /** specifies a callback function for when the value in `SearchField` is changed by the user. The default is an empty function. For example:
+  ```jsx
+  <SearchField onChange={value => console.log(value)} />
+  ``` */
+  onChange: PropTypes.func,
+  /** specifies a callback function for when the value in `SearchField` is cleared by the user. The default is an empty function. For example:
+  ```jsx
+  <SearchField onClear={() => console.log('search cleared')} />
+  ``` */
+  onClear: PropTypes.func,
+  /** specifies a callback function for when the user focuses in the `SearchField` component. The default is an empty function. For example:
+  ```jsx
+  <SearchField onFocus={event => console.log(event)} />
+  ``` */
+  onFocus: PropTypes.func,
+  /** specifies the screenreader text for both the clear and submit buttons (e.g., for i18n translations). The default is `{ label: 'search', clearButton: 'clear search', searchButton: 'submit search' }`. */
+  screenReaderText: PropTypes.shape({
+    label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
+    submitButton: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
+    clearButton: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  }),
+  /** specifies the initial value for the input. The default is an empty string. */
+  value: PropTypes.string,
+  /** specifies the icon element(s) to use for the clear and submit buttons. The default is `{ submit: <FontAwesomeIcon icon={faSearch} />, clear: <FontAwesomeIcon icon={faTimes} /> }`. */
+  icons: PropTypes.shape({
+    submit: PropTypes.element.isRequired,
+    clear: PropTypes.element,
+  }),
+  /** specifies the aria-label attribute on the form element. This is useful if you use the `SearchField` component more than once on a page. */
+  formAriaLabel: PropTypes.string,
+};
+
+SearchFieldAdvanced.defaultProps = {
+  className: undefined,
+  formAriaLabel: undefined,
```

Comment:
What do you think about using just `aria-label` as the prop. As it stands the prop name kind of forces you to know a little about the inner-workings of the SearchField. ..but it could be counterintuitive if we're putting the label on a particular element not the wrapper. Curious your thoughts.

Reply:
Yeah, I was also deciding between just `aria-label` and something more specific to the form element (i.e., `formAriaLabel`). I opted for `formAriaLabel` because it does help describe which element the `aria-label` applies to. I'm still leaning towards the more explicit prop name despite it forcing you to know a bit about the inner-workings of the component.

The only time this prop would be used is if there are multiple instances of the `SearchField` component on the same page. The `aria-label` on the form can be used to differentiate the search landmarks.